lib/thredded/per_notifier_pref.rb

```diff
@@ -0,0 +1,102 @@
+# frozen_string_literal: true
+module Thredded
+  # in this class we dynamically set up an accessor for all the Thredded.notifiers (by their key)
+  # as we don't know in advance what the keys are,
+  # we try to not pollute the namespace
+  # nonetheless some keys would be bad, so we sanity check those on setup
+  class PerNotifierPref
```

Comment:
I think the complexity of this approach lies mostly in providing both the database model and the view for the plugins. This makes writing plugins easier but limits what the plugins can do and makes the core more complex.

Consider also if perhaps a plugin should just add columns for its own settings and hook into the settings view via view?